src/Umbraco.Web/Editors/MacroController.cs

```diff
@@ -49,7 +49,28 @@ public IEnumerable<MacroParameter> GetMacroParameters(int macroId)
         /// 
         /// </param>
         /// <returns></returns>
-        public HttpResponseMessage GetMacroResultAsHtmlForEditor(string macroAlias, int pageId, [FromUri]IDictionary<string, object> macroParams)
+        [HttpGet]
+        public HttpResponseMessage GetMacroResultAsHtmlForEditor(string macroAlias, int pageId, [FromUri] IDictionary<string, object> macroParams)
+        {
+            return GetMacroResultAsHtml(macroAlias, pageId, macroParams);
+        }
+
+        /// <summary>
+        /// Gets a rendered macro as html for rendering in the rich text editor.
+        /// Using HTTP POST instead of GET allows for more parameters to be passed as it's not dependant on URL-length limitations like GET.
+        /// The method using GET is kept to maintain backwards compatibility
+        /// </summary>
+        /// <param name="macroAlias"></param>
+        /// <param name="pageId"></param>
+        /// <param name="macroParams"></param>
+        /// <returns></returns>
+        [HttpPost]
+        public HttpResponseMessage GetMacroResultAsHtmlForEditorUsingHttpPost(string macroAlias, int pageId, [FromBody]IDictionary<string, object> macroParams)

```

Comment:
I'll fix this up but this should remain the same name as the other action method and accept a model instead of parameters.